sql/mozfun/ltv/get_state_ios_v2/udf.sql

```diff
@@ -0,0 +1,56 @@
+CREATE OR REPLACE FUNCTION ltv.get_state_ios_v2(
+  days_since_first_seen INT64,
+  days_since_seen INT64,
+  submission_date DATE,
+  death_time INT64,
+  pattern INT64,
+  active INT64,
+  max_weeks INT64
+)
+RETURNS STRING AS (
+   -- users age. 0 on their first day (we need this uniquely ID'd) and then number of the week (1-indexed) since they were new.
+   -- their second week starts on day 8, etc.
+  CONCAT(
+    CASE
+      WHEN days_since_first_seen = 0
+        THEN 0
+      WHEN (days_since_first_seen != 0)
+        AND (((days_since_first_seen + 1) / 7) < max_weeks)
+        THEN CEILING((days_since_first_seen + 1) / 7)
+      ELSE max_weeks
+    END,
+    '_dow',
+     -- day of week. allows the model to capture weekly seasonality
+    EXTRACT(DAYOFWEEK FROM submission_date),
+    '_',
+     -- `pattern` is the number of active days in the last 28 days.
+     -- users are binned into equal sized activity levels. 0 would be inactive.
+    CASE
+      WHEN days_since_seen >= death_time
+        THEN '00'
+      WHEN pattern
+        BETWEEN 1
+        AND 7
+        THEN '1'
+      WHEN pattern
+        BETWEEN 8
+        AND 14
+        THEN '2'
+      WHEN pattern
+        BETWEEN 15
+        AND 21
+        THEN '3'
+      WHEN pattern > 21
+        THEN '4'
+      ELSE CAST(pattern AS string)
```

Comment:
Should we have a default value in case of nulls? Trying this in the console it appears if one of the values inside the concat is null the whole return values becomes null:
```
SELECT CONCAT("test", CAST(NULL AS STRING))
UNION ALL
SELECT CONCAT("test", "_", "test2", CAST(NULL AS STRING))
```
returns:
```
null
null
```

Reply:
@irrationalagent As the UDF stands, if any input value is null, the whole result is null - is this intended or do you want any logic added for null handling?  Or are all of these values always not null?

Reply:
The way we have been handling this is to manage the input so that it is never null (i.e. by using COALESCE, or just filtering out nulls in the input dataset). Let's continue to do that - these functions aren't really meant for use outside of the datasets that we tailor for the input.

Reply:
Sounds good, thanks for confirming!